Apps/Src/CommandLine.c

```diff
@@ -0,0 +1,357 @@
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include "os.h"
+#include "Tasks.h"
+#include "BSP_UART.h"
+#include "CANbus.h"
+#include "Contactors.h"
+#include "Minions.h"
+#include "Pedals.h"
+
+#define MAX_BUFFER_SIZE	128	// defined from BSP_UART_Read function
+
+// Represents a command that the command line understands
+struct Command {
+	const char *name;
+	bool (*action)(void);
+};
+
+static bool cmd_help(void);
+
+static bool cmd_CANbus_Send(void);
+
+static bool cmd_CANbus_Read(void);
+
+static bool cmd_Contactors_Get(void);
+
+static bool cmd_Contactors_Set(void);
+
+static bool cmd_Minion_Read_Input(void);
+
+static bool cmd_Minion_Write_Output(void);
+
+static bool cmd_Pedals_Read(void);
+
+
+const struct Command cmdline_commands[] = {
+	{.name = "help", .action = cmd_help},
+	{.name = "CANbus_Send", .action = cmd_CANbus_Send},
+	{.name = "CANbus_Read", .action = cmd_CANbus_Read},
+	{.name = "Contactors_Get", .action = cmd_Contactors_Get},
+	{.name = "Contactors_Set", .action = cmd_Contactors_Set},
+	{.name = "Minion_Read_Input", .action = cmd_Minion_Read_Input},
+	{.name = "Minion_Write_Output", .action = cmd_Minion_Write_Output},
+	{.name = "Pedals_Read", .action = cmd_Pedals_Read},
+	{.name = NULL, .action = NULL}
```

Comment:
But then it might be confused as a command with (currently) no action. I think it's better to leave the sentinel as all NULL. But it's probably a good idea to leave a comment specifically saying it's a sentinel.